packages/core/src/parsers/linkProcessor.js

```diff
@@ -23,41 +41,101 @@ const defaultTagLinkMap = {
  * @param {string} baseUrl
  */
 function convertRelativeLinks(node, cwf, rootPath, baseUrl) {
-  if (!(node.name in defaultTagLinkMap)) {
-    return;
-  }
-
-  const linkAttribName = defaultTagLinkMap[node.name];
-  const resourcePath = node.attribs && node.attribs[linkAttribName];
+  const resourcePath = getResourcePath(node, cwf, rootPath);
   if (!resourcePath) {
     return;
   }
-
   if (path.isAbsolute(resourcePath) || utils.isUrl(resourcePath) || resourcePath.startsWith('#')) {
     // Do not rewrite.
     return;
   }
 
   const cwd = path.dirname(cwf);
   const fullResourcePath = path.join(cwd, resourcePath);
-  const resourcePathFromRoot = utils.ensurePosix(path.relative(rootPath, fullResourcePath));
+  const resourcePathFromRoot = getResourcePathFromRoot(rootPath, fullResourcePath);
 
+  const linkAttribName = defaultTagLinkMap[node.name];
   node.attribs[linkAttribName] = path.posix.join(baseUrl || '/', resourcePathFromRoot);
 }
-
-function convertMdExtToHtmlExt(node) {
+function convertMdAndMbdExtToHtmlExt(node) {
   if (node.name === 'a' && node.attribs && node.attribs.href) {
+    const hasNoConvert = lodashHas(node.attribs, 'no-convert');
+    if (hasNoConvert) {
+      return;
+    }
     const { href } = node.attribs;
     const hasMdExtension = href.slice(-3) === '.md';
-    const hasNoConvert = lodashHas(node.attribs, 'no-convert');
-    if (hasMdExtension && !hasNoConvert) {
+    if (hasMdExtension) {
       const newHref = `${href.substring(0, href.length - 3)}.html`;
       node.attribs.href = newHref;
+      return;
+    }
+    const hasMbdExtension = href.slice(-4) === '.mbd';
+    if (hasMbdExtension) {
+      const newHref = `${href.substring(0, href.length - 4)}.html`;
+      node.attribs.href = newHref;
+    }
+  }
+}
+
+function validateIntraLink(node, cwf, config) {
+  // checks if the file indicated in the intra link exists
+  let resourcePath = getResourcePath(node, cwf, config.rootPath);
+  if (!resourcePath || utils.isUrl(resourcePath) || resourcePath.startsWith('#')) {
+    return;
+  }
+
+  // remove hash portion (if any) in the resourcePath
+  const resourcePathUrl = url.parse(resourcePath);
+  if (resourcePathUrl.hash) {
+    resourcePath = resourcePathUrl.path;
+  }
+
+  // check if the file is a Page source
+  // since this function is called after convertMdAndMbdExtToHtml,
+  // we can assume that page source mostly has .html extension (unless user indicates no-convert)
+  const hasHtmlExt = resourcePath.slice(-5) === '.html';
+  const hasMdExt = resourcePath.slice(-3) === '.md'; // suppose user indicates no-convert
+  const hasMbdExt = resourcePath.slice(-4) === '.mbd'; // suppose user indicates no-convert
```

Comment:
> we can be sure here that they are pointing to assets, because pages always generate `.html`s

Hmm, I'm pretty confused here. If I understand you correctly, you mean that pages (regardless .mb or .mbd), always generates and outputs `.html`s to the `_site` folder, right? I am not sure how 'pages always generate .html' comes into the picture for the current context. 

My thought process is this: 
1. User has intra-link in document which can be of `.md, .mbd, or .html` extensions (however, they are mostly .html extension in our current user docs and dev docs, due to past user behaviors, as we did not have the auto conversion back then)
2. After converting the intra-links using `convertMdAndMbdExtToHtmlExt`, the intra-links can still be one of `.md, .mbd, or .html` extensions, right? We can have `.md or .mbd` extensions in the scenario where the user indicates the `no-convert` attribute in their tag. That was why I thought we have to check for those extensions here. If we just check for `.html` extension, we won't be able to validate the intra-links that have `.md or .mbd` extensions. 

Regarding your point about pages always generating and outputting `.html` into `_site`, how would that make the `.md and .mbd` extension checks redundant? Or rather, how are they related? I can't seem to draw the relation :-/

Reply:
if pages always generate `.html`, then we know that `.md` / `.mbd` links would definitely be pointing to assets already.

because by default we assume such links are pointing to pages (hence converting them `.html`).
if there is still a `.md` / `.mbd` link post conversion (indicated via `no-convert`), that means the user wants to point to a raw asset file (e.g. the common case of `readme.md`)

Reply:
> .md / .mbd links would definitely be pointing to assets already.

Hmm, let me clarify my basic understanding. By this, you mean that `.md` and `.mbd` links would be pointing to the generated `.html` files right? That is why we convert the `.md` and `.mbd` extensions to `.html` extensions. 

> if there is still a .md / .mbd link post conversion (indicated via no-convert), that means the user wants to point to a raw asset file (e.g. the common case of readme.md)

Hmm, so you mean that for the case of `readme.md`, we will fall back to the asset fs check? I noticed that the current ignore config that we have, `*.md` files are ignored. This means that we wouldn't have to check for the existence of `readme.md` files, right? 


Also, I think the `readme.md` would fall under the addressablePages, right? So if we want to check for its existence, we can either check under the 1. Pages check step or 2. the asset fs check, right?

Reply:
> Hmm, let me clarify my basic understanding. By this, you mean that .md and .mbd links would be pointing to the generated .html files right? That is why we convert the .md and .mbd extensions to .html extensions.

yup 👍 

> I noticed that the current ignore config that we have, *.md files are ignored. This means that we wouldn't have to check for the existence of readme.md files, right?

it may not be ignored in all cases; the solution should be generalizable to anything that the user can configure (not just our docs)

> Also, I think the readme.md would fall under the addressablePages, right?

not if it is excluded (or just not included in any glob / src)
https://markbind.org/userGuide/siteJsonFile.html#pagesexclude